### ISSUE TITLE
no-useless-initializer: false-negative with destructuring of tuple types

### ISSUE THREAD

**Issue by ajafff** [fimbullinter/wotan#441]
```ts
let [a = '', b = 1, c = true, d = null] = get<[string, number, boolean, object?]>();
```

Current behavior: no error
Expected behavior: error on `a = ''`, `b = 1` and `c = true`;
